File: src/bdd.py
from prisma import PrismaClient
from access_token import get_access_token
from requests import search_playlist_id, search_playlist
import sqlite3

# Obtenez le token
access_token = get_access_token()

# Initialisez le client Prisma
prisma = PrismaClient()

playlist_name = ["Pop mix", "Rock", "Années 80", "Années 90", "Musique de film", "Jazz", "R&B", "Country", "Dubstep",
                 "Hip-Hop", "Rap", "Rap Francais", "Rap US", "Beurette à chicha", "Funk", "Années 2000", "Afrotrap",
                 "Bachata", "Latino", "Dua Lipa"]

for theme_only in playlist_name:
    # Effectuez la recherche de la playlist
    result_playlist_id = search_playlist_id(access_token, theme_only)

    for playlist_info in result_playlist_id.get("playlists", {}).get("items", []):
        playlist_id = playlist_info.get("id")
    print(playlist_id)

    result_playlist = search_playlist(access_token, str(playlist_id))

    # Connect to the database
    conn = sqlite3.connect("spotify_tracks.db")
    cursor = conn.cursor()

    # Create the table if it doesn't exist
    cursor.execute('''
        CREATE TABLE IF NOT EXISTS tracks (
            track_id TEXT PRIMARY KEY,
            name TEXT,
            artist TEXT,
            preview_url TEXT,
            image_url TEXT,
            theme TEXT
        )
    ''')

    # Loop through each track in the playlist
    for track_info in result_playlist.get("tracks", {}).get("items", []):
        # Get the track information
        track_id = track_info.get("track", {}).get("id")
        track_name = track_info.get("track", {}).get("name")
        track_artist = track_info.get("track", {}).get("artists", [{}])[0].get("name")
        preview_url = track_info.get("track", {}).get("preview_url")
        image_url = track_info.get("track", {}).get("album", {}).get("images", [{}])[0].get("url")

        if preview_url is not None:
            # Utilisez le client Prisma pour insérer ou mettre à jour la base de données
            prisma.track.upsert(
                where={"track_id": track_id},
                create={"name": track_name, "artist": track_artist, "preview_url": preview_url,
                        "image_url": image_url, "theme": theme_only},
                update={"name": track_name, "artist": track_artist, "preview_url": preview_url,
                        "image_url": image_url, "theme": theme_only},
            )

            print(f"Nom de la piste: {track_name}")
            print(f"Artiste: {track_artist}")
            print(f"ID Spotify: {track_id}")
            print(f"URL de prévisualisation: {preview_url}")
            print(f"URL de l'image: {image_url}")
            print("\n")

    # Commit the changes to the database
    conn.commit()

    # Close the connection to the database
    conn.close()

# Fermez la connexion à la base de données Prisma
prisma.disconnect()

File: prisma/bdd.ts
import { PrismaClient } from '@prisma/client';
import { getAccessToken } from '../src/access_token'; // Assurez-vous d'ajuster le chemin du fichier access_token
import { searchPlaylistId, searchPlaylist } from '../src/requests'; // Assurez-vous d'ajuster le chemin du fichier requests

// Obtenez le token
const accessToken = getAccessToken();

// Initialisez le client Prisma
const prisma = new PrismaClient();

const playlistNames: string[] = [
    "Pop mix", "Rock", "Années 80", "Années 90", "Musique de film", "Jazz", "R&B", "Country", "Dubstep",
    "Hip-Hop", "Rap", "Rap Francais", "Rap US", "Beurette à chicha", "Funk", "Années 2000", "Afrotrap",
    "Bachata", "Latino", "Dua Lipa"
];

for (const themeOnly of playlistNames) {
    // Effectuez la recherche de la playlist
    const resultPlaylistId = searchPlaylistId(accessToken.toString(), themeOnly);

    for (const playlistInfo of resultPlaylistId.playlists?.items || []) {
        const playlistId = playlistInfo?.id;
        console.log(playlistId);

        const resultPlaylist = searchPlaylist(accessToken.toString(), playlistId?.toString() || '');

        // Loop through each track in the playlist
        for (const trackInfo of resultPlaylist.tracks?.items || []) {
            // Get the track information
            const trackId = trackInfo.track?.id;
            const trackName = trackInfo.track?.name;
            const trackArtist = trackInfo.track?.artists?.[0]?.name;
            const previewUrl = trackInfo.track?.preview_url;
            const imageUrl = trackInfo.track?.album?.images?.[0]?.url;

            if (previewUrl) {
                // Utilisez le client Prisma pour insérer dans la base de données
                prisma.track.create({
                    data: {
                        track_id: trackId || '',
                        name: trackName || '',
                        artist: trackArtist || '',
                        preview_url: previewUrl || '',
                        image_url: imageUrl || '',
                        theme: themeOnly || '',
                    },
                });

                console.log(`Nom de la piste: ${trackName}`);
                console.log(`Artiste: ${trackArtist}`);
                console.log(`ID Spotify: ${trackId}`);
                console.log(`URL de prévisualisation: ${previewUrl}`);
                console.log(`URL de l'image: ${imageUrl}`);
                console.log("\n");
            }
        }
    }
}

// N'oubliez pas de fermer la connexion Prisma à la fin de votre script
prisma.$disconnect();
